forms/admission.py

```diff
@@ -101,3 +103,59 @@ class Meta:
             'state',
             'state_reason'
         ]
+
+
+class RejectedAdmissionForm(ModelForm):
+
+    rejected_reason = forms.ChoiceField(
+        choices=REJECTED_REASON_CHOICES,
+        required=True,
+        label=_('Predefined reason'),
+    )
+    other_reason = forms.CharField(
+        required=False,
+        label=_('Other rejection reason'),
+    )
+
+    class Meta:
+        model = Admission
+        fields = [
+            'state',
+            'state_reason',
+        ]
+
+    def __init__(self, data, **kwargs):
+
+        super().__init__(data, **kwargs)
+
+        if data is None:
+            # GET
+            if self.instance.state == admission_state_choices.REJECTED:
+                self._reject_state_init()
+            else:
+                self._disabled_and_init_other_reason()
+
+        self.fields['rejected_reason'].widget.attrs = {'onchange': 'disabledEnabledOtherReasonInputText();'}
+
+    def _reject_state_init(self):
+        if any(self.instance.state_reason in reason for reason in REJECTED_REASON_CHOICES):
```

Comment:
:+1: super code :D 
Peut-être juste un petit truc : là on regarde si la string `self.instance.state_reason` est une sous-string de la string `reason` (code : `self.instance.state_reason in reason`).
Je pense qu'on peut se contenter de checker l'égalité : 
` if any(self.instance.state_reason == reason for reason in REJECTED_REASON_CHOICES)`